src/fallback/implementation.cpp

```diff
@@ -350,15 +350,24 @@ simdutf_warn_unused size_t implementation::maximal_binary_length_from_base64(con
 }
 
 simdutf_warn_unused result implementation::base64_to_binary(const char * input, size_t length, char* output) const noexcept {
-  return scalar::base64::base64_to_binary(input, length, output);
+  if(length > 0 && input[length - 1] == '=') {
+    length -= 1;
+    if(length > 0 && input[length - 1] == '=') {
+      length -= 1;
+    }
+  }
+  if(length == 0) {
```

Comment:
If we move this if statement before the previous line, the implementation can be simplified a lot

Reply:
Can you elaborate? I am sure that this function can be simplified, but I am not sure what you have in mind. The trick here is that we must discard the padding characters before we can have the 'true' length. Here we decide that if the empty is empty, or contains only one or two padding characters ('='), then we accept it as valid, but return 0 bytes.

Reply:
Actually yes you're right. Resolving this.